bindings/rust/src/lib.rs

```diff
@@ -40,14 +40,66 @@ mod sys;
 use std::ffi::CString;
 use std::ptr::NonNull;
 
+/// A safe container for handling the low-level FizzyError struct.
+struct FizzyErrorBox(Box<sys::FizzyError>);
+
+impl FizzyErrorBox {
+    /// Create a safe, boxed, and zero initialised container.
+    fn new() -> Self {
+        FizzyErrorBox {
+            0: Box::new(sys::FizzyError {
+                code: 0,
```

Comment:
can you use here `sys::FIZZY_SUCCESS`?

Reply:
I prefer keeping zeroes, because all I want to do here is zero initialised it, i.e. `memset(&err, 0, sizeof(err))`.